src/callbacks_stage/subcell_bounds_check_2d.jl

```diff
@@ -5,7 +5,8 @@
 @muladd begin
 #! format: noindent
 
-@inline function check_bounds(u, equations, solver, cache,
+@inline function check_bounds(u::AbstractArray{<:Any, 4},
```

Comment:
Why is this dispatch needed if you didn't add a new `function check_bounds`?

Reply:
I just wanted to make sure that an error is thrown when calling this routine with another dimension than 2. In the past, I had `mesh::Union{TreeMesh2D, StructuredMesh{2}}` here, but since I didn't need the mesh in here, I deleted the mesh as parameter.
But I don't have a strong opinion on this and can also delete it again.

Reply:
Alright, it's fine like this for me! :)